src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1131,6 +1131,12 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
                     }
                     throw;
                 }
+                catch (NpgsqlException e) when (readingNotifications2 && e.InnerException is TimeoutException timeout)
```

Comment:
Why is this special catch needed? Does the catch-all below not work for some reason?

Reply:
Mostly, so there are no changes for `NpgsqlConnector.Wait`. But thinking on it again, checking the `InnerException` there might be a much better solution.